src/Http/Search/SearchQuery.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Werkspot\KvkApi\Http\Search;
+
+final class SearchQuery implements QueryInterface
+{
+
+    /**
+     * @info KvK number, identifying number for a registration in the Netherlands Business Register. Consists of 8 digits
+     * @var string
+     */
+    private $kvkNumber;
+
+    /**
+     * @info Branch number (Vestigingsnummer), identifying number of a branch. Consists of 12 digits
+     * @var string
+     */
+    private $branchNumber;
+
+    /**
+     * @info RSIN is an identification number for legal entities and partnerships. Consist of only digits
+     * @var int
+     */
+    private $rsin;
+
+    /**
+     * @info Street of an address
+     * @var string
+     */
+    private $street;
+
+
+    /**
+     * @info House number of an address
+     * @var string
+     */
+    private $houseNumber;
+
+    /**
+     * @info Postal code or ZIP code, example 1000AA
+     * @var string
+     */
+    private $postalCode;
+
+    /**
+     * @info City or Town name
+     * @var string
+     */
+    private $city;
+
+
+    /**
+     * @info Indication  to include searching through inactive dossiers/deregistered companies.
+     * @note History of inactive companies is after 1 January 2012
+     * @var bool
+     */
+    private $includeInactiveRegistrations;
+
+    /**
+     * @info restrictToMainBranch Search is restricted to main branches.
+     * @var bool
+     */
+    private $restrictToMainBranch;
+
+    /**
+     * @info Defines the search collection for the query
+     * @var string
+     */
+    private $site;
+
+    /**
+     * @info User can optionally add a context to identify his result later on
+     * @var string
+     */
+    private $context;
+
+    /**
+     * @info Free format text search for in the compiled search description.
+     * @var string
+     */
+    private $q;
+
+
+    /**
+     * @return string
+     */
+    public function getStreet(): string
+    {
+        return $this->street;
+    }
+
+    /**
+     * @param string $street
+     */
+    public function setStreet(string $street): void
+    {
+        $this->street = $street;
```

Comment:
Since this class is not just a Query, but also a query **builder**, would it be nice to design it as a builder class making use of a fluent interface, as
```php
    public function setStreet(string $street): SearchQuery
    {
        $this->street = $street;

        return $this;
    }
```
?

Reply:
For similarity it must also be changed than in the ProfileQuery than.

Reply:
For consistency, yes. However, we will not require you to do that change for this MR to be merged, only that you do it in your PR.

I will take care of removing it in the older code, although then we will need to bump up the major version, so it doesn't break the projects using that code. We will do that later, together with other BC breaking changes.